### PR TITLE
[example/nrfconnect] Added support for sending UDP broadcast messages.

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
#### Problem
nrfconnect platform doesn't support sending UDP broadcast messages periodically as other platforms do (e.g. nrf5)

 #### Summary of Changes
* Added Service component for realizing UDP broadcast support.

### CODE REVIEW

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 1]
```diff
@@ -179,6 +181,15 @@ int AppTask::StartApp()
         sStatusLED.Animate();
         sUnusedLED.Animate();
         sUnusedLED_1.Animate();
+
+        uint64_t nowUS            = chip::System::Platform::Layer::GetClock_Monotonic();
+        uint64_t nextChangeTimeUS = mLastChangeTimeUS + 5 * 1000 * 1000UL;
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:186]: Make 5 * 1000 * 1000 a constant?

File: examples/lighting-app/nrfconnect/main/AppTask.cpp [hunk 2]
```diff
@@ -105,6 +106,7 @@ int AppTask::Init()
 int AppTask::StartApp()
 {
     int ret = Init();
+    uint64_t mLastChangeTimeUS = 0;
```
Comment (Damian-Nordic) [inline examples/lighting-app/nrfconnect/main/AppTask.cpp:109]: mLastServiceBroadcastTimeUS?

File: examples/platform/nrfconnect/app/Service.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+// Transport Callbacks
+namespace {
+
+char deviceName[128];
+constexpr uint16_t kUDPBroadcastPort = 23367;
+
+} // namespace
+
+void SetDeviceName(const char * newDeviceName)
```
Comment (Damian-Nordic) [inline examples/platform/nrfconnect/app/Service.cpp:49]: This doesn't seem to be used. How about using ConnectivityMgr().GetBLEDeviceName()? deviceName array could be of size CONFIG_BT_DEVICE_NAME_MAX then.

Also, we might actually set BLEDeviceName() to "CHIP Lock"/"CHIP Lighting" or so (though we may do that in a separate commit.

File: examples/platform/nrfconnect/app/Service.cpp [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    {
+        return;
+    }
+    chip::DeviceLayer::ThreadStackMgrImpl().LockThreadStack();
+    otError error = OT_ERROR_NONE;
+    otMessageInfo messageInfo;
+    otUdpSocket mSocket;
+    otMessage * message = nullptr;
+
+    memset(&mSocket, 0, sizeof(mSocket));
```
Comment (Damian-Nordic) [inline examples/platform/nrfconnect/app/Service.cpp:67]: Note that in c++11 you can initialize local objects in the following way:
otMessageInfo messageInfo = {};

**Review by Damian-Nordic — APPROVED** [review]
Looks good, just a few suggestions

File: examples/lighting-app/nrfconnect/main/AppTask.cpp (hunk 1 above)
Comment (kkasperczyk-no) [reply examples/lighting-app/nrfconnect/main/AppTask.cpp]: Done

File: examples/lighting-app/nrfconnect/main/AppTask.cpp (hunk 2 above)
Comment (kkasperczyk-no) [reply examples/lighting-app/nrfconnect/main/AppTask.cpp]: Done

File: examples/platform/nrfconnect/app/Service.cpp (hunk 3 above)
Comment (kkasperczyk-no) [reply examples/platform/nrfconnect/app/Service.cpp]: For now I removed this method. I will set device names in the separate PR.
